### PR TITLE
Remove redundant direct router library dependency

### PR DESCRIPTION
**Pull request by tsiren**
Signed-off-by: Tuomo Sirén <tuomo.siren@gmail.com>

## Description

Remove direct dependency to gorilla-mux. Standard library provides routing capabilities in itself thus there is no need for additional routing library unless doing e.g. regexp based routing. In this case avoiding a separate library eases debugging and makes the source code walk through more pleasant.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [ ] Manual test
- [x] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,3 +22,4 @@ static/zz_generated_assets.go
 docs/cli
 site
 k0sctl.yaml
+*.swp
```
Comment (twz123) [inline .gitignore:25]: I'd suggest to put those things into `$XDG_CONFIG_HOME/git/ignore`, so we don't have to accumulate all sorts of OS and/or editor specific stuff here.

Comment (tsiren) [reply .gitignore]: Removed changes to this file

**Comment by twz123** [comment]
Thanks @tsiren. Could you please add your signoff to 9d4e016b1357de15f60a333a64b8ee0feb1d19e4?

File: .gitignore (hunk 1 above)
Comment (tsiren) [reply .gitignore]: Added signoff to latest commit.

**Review by makhov — APPROVED** [review]

**Pull request merged**